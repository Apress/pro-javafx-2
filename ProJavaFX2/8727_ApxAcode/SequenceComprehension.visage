/*
 * @test
 * @run
 */

var seq = for (x in [1..4]) x*x;
print("seq = "); println(seq);

seq = for (x in [1..4] where x > 2) x*x;
print("seq = "); println(seq);

var rows = ["A", "B"];
var columns = [1, 2];
var matrix = for (row in rows, column in columns) "{row}{column}";
print("matrix = "); println(matrix);

var digits = [1, 2, 3];
var seq1 = for (x in digits, y in digits) "{x}{y}";
print("seq1 = "); println(seq1);

var seq2 = for (x in digits where x > 1, y in digits where y >= x) {
  "{x}{y}"
}
print("seq2 = "); println(seq2);
