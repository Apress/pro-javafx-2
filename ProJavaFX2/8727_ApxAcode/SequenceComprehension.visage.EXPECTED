seq = [ 1, 4, 9, 16 ]
seq = [ 9, 16 ]
matrix = [ A1, A2, B1, B2 ]
seq1 = [ 11, 12, 13, 21, 22, 23, 31, 32, 33 ]
seq2 = [ 22, 23, 33 ]
